### PR TITLE
ginkgo: adding coveralls to the CI

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6094)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: ginkgo.Jenkinsfile [hunk 1]
```diff
@@ -64,6 +65,7 @@ pipeline {
             }
             steps {
                 sh "cd ${TESTDIR}; make tests-ginkgo"
+                sh "go get golang.org/x/tools/cmd/cover && go get github.com/mattn/goveralls && find -iname "*goveralls*" && $HOME/go/bin/goveralls -coverprofile=${TESTDIR}coverage-all.out -service=travis-ci -repotoken=${COVERALLS_TOKEN}"
```
Comment (eloycoto) [inline ginkgo.Jenkinsfile]: I know that is wip, but:
 
- Could we use a docker container?
- Due we run the test using docker-compose, could we create a target in `test/docker-compose.yaml`

Regards

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: test/docker-compose.yml [hunk 2]
```diff
@@ -27,3 +27,9 @@ services:
     extends:
       service: base_image
     command: "bash -c 'cd /go/src/github.com/cilium/cilium/; make precheck || exit 1'"
+  publish_unit_test_cover:
+    extends:
+      service: base_image
+    environment:
+      - "COVERALLS_TOKEN"
```
Comment (eloycoto) [inline test/docker-compose.yml]: `- "COVERALLS_TOKEN=${COVERALLS_TOKEN}"`

File: test/docker-compose.yml [hunk 3]
```diff
@@ -27,3 +27,9 @@ services:
...
+    command: "bash -c 'cd /go/src/github.com/cilium/cilium/; go get golang.org/x/tools/cmd/cover && go get github.com/mattn/goveralls && goveralls -coverprofile=coverage-all.out -service=travis-ci -repotoken=${COVERALLS_TOKEN}'"
```
Comment (eloycoto) [inline test/docker-compose.yml]: TBH, the idea of using Docker-compose is to have the use of a docker-image and do not install in each build that tools.

**Comment by aanm** [comment]
test-me-please

File: test/docker-compose.yml (hunk 3 above)
Comment (aanm) [reply test/docker-compose.yml]: this is still WIP, I'll create an image once I know this is working

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by eloycoto** [comment]
About the docker-compose: 

```
brea.acalustra.com $ git diff docker-compose.yml | tee
diff --git a/test/docker-compose.yml b/test/docker-compose.yml
index 11ba7e381..67b08f8d4 100644
--- a/test/docker-compose.yml
+++ b/test/docker-compose.yml
@@ -11,6 +11,9 @@ services:
     hostname: "etcd"
     command: "etcd -name etcd0 -advertise-client-urls http://0.0.0.0:4002 -listen-client-urls http://0.0.0.0:4002 -initial-cluster-token etcd-cluster-1 -initial-cluster-state new"
     privileged: true
+  cov:
+    image: "golang:1.10"
+    command: "/bin/echo FLAG=${MY_FLAG}"
   base_image:
     image: "quay.io/cilium/cilium-builder:2018-09-17"
     volumes:
brea.acalustra.com $ export MY_FLAG="21399"
brea.acalustra.com $ docker-compose -f docker-compose.yml run cov
FLAG=21399
brea.acalustra.com $ export MY_FLAG=""
brea.acalustra.com $ docker-compose -f docker-compose.yml run cov
FLAG=
brea.acalustra.com $ export MY_FLAG="test"
brea.acalustra.com $ docker-compose -f docker-compose.yml run cov
FLAG=test
brea.acalustra.com $

```

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by eloycoto — CHANGES_REQUESTED** [review]
I would disable the unitest on the ginkgo.Jenkinsfile.

**Comment by aanm** [comment]
@eloycoto let's keep it because if the unit tests fails we won't run the other CI tests.

**Review by jrajahalme — APPROVED** [review]
Reviewing for proxy only

**Review by eloycoto — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Pull request merged**